homework_9/exceptionsHw9/ArrayValueCalculator.java

```diff
@@ -0,0 +1,21 @@
+package exceptionsHw9;
+
+public class ArrayValueCalculator {
+    public static int doCalc(String[][] matrix) throws ArraySizeException, ArrayDataException {
```

Comment:
`throws ArraySizeException, ArrayDataException` should be used if you wanna delegate error handling to the end user. The description asks you to validate the input on your own, so `throws` is redundant here, remove it.